### ISSUE TITLE
Multi-index Triptych

### ISSUE THREAD

**Issue by SarangNoether** [monero-project/research-lab#66]
The [Triptych](https://eprint.iacr.org/2020/018) proving system is a `d`-LSAG-type construction with functionality similar to that of MLSAG or CLSAG, permitting proofs on multiple components of key vectors in a set while hiding the signing index. The version specified in the preprint can be integrated into the Monero transaction protocol, requiring a separate proof for each spent input and using commitment offsets to assert transaction balance.

However, we've investigated an alternate Triptych construction (in [math](https://github.com/SarangNoether/skunkworks/blob/triptych/paper/multi.tex) and [code](https://github.com/SarangNoether/skunkworks/tree/triptych/triptych-multi)) that permits signing for multiple indices in the same list simultaneously within the same proof, while also taking care of balance assertion.

However, the soundness of this extension to Triptych is unknown. If shown to reduce to a known cryptographic hardness assumption, it would be useful due to its scaling benefits.

**Comment by UkoeHB** [comment]
Would this imply all inputs are at the same index? IIRC RCTTypeFull was never really used much due to that problem.

**Comment by SarangNoether** [comment]
No, the opposite. There is a single list of commitments, and a single proof is used for all indices in the list corresponding to all signers at once.
